lib/replicate/checkpointer.js

```diff
@@ -64,13 +94,43 @@ Checkpointer.prototype.updateSource = function (checkpoint) {
     });
 };
 
+var comparisons = {
+  "undefined": function(targetDoc, sourceDoc) {
+    // This is the previous comparison function
+    if (collate(targetDoc.last_seq, sourceDoc.last_seq) === 0) {
+      return sourceDoc.last_seq;
+    }
+
+    return 0;
+  },
+  "2": function(targetDoc, sourceDoc) {
+    // This is the comparison function ported from CouchDB
+    return compareReplicationLogs(sourceDoc, targetDoc).last_seq;
+  }
+};
+
 Checkpointer.prototype.getCheckpoint = function () {
   var self = this;
   return self.target.get(self.id).then(function (targetDoc) {
     return self.src.get(self.id).then(function (sourceDoc) {
-      if (collate(targetDoc.last_seq, sourceDoc.last_seq) === 0) {
-        return sourceDoc.last_seq;
+      // TODO: Discuss the 'return 0's here
+      // can we do something better?
```

Comment:
The best way would be to migrate the doc which is not version 2 in the case they are not the same and then use the comparison for 2.